### PR TITLE
Add mypy-plugin extra dependency

### PR DESCRIPTION
**Pull request by zhukovgreen**
use `pip install classes[compatible-mypy]` for installing additional dependencies required by the mypy-plugin.

# I have made things!

## Checklist

- [x] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [ ] I have created at least one test case for the changes I have made
- [x] I have updated the documentation for the changes I have made
- [ ] I have added my changes to the `CHANGELOG.md`

## Related issues

🙏 Please, if you or your company finds `dry-python` valuable, help us sustain the project by sponsoring it transparently on https://github.com/sponsors/dry-python. As a thank you, your profile/company logo will be added to our main README which receives hundreds of unique visitors per day.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -48,6 +48,7 @@ _ = "returns.contrib.hypothesis._entrypoint"
 python = "^3.7"
 
 typing-extensions = ">=4.0,<5.0"
+mypy = { version = "^1.2.0", optional = true }
```
Comment (sobolevn) [inline pyproject.toml:51]:
```suggestion
mypy = { version = "^1.2", optional = true }
```

**Review by sobolevn — COMMENTED** [review]
I would prefer this name: https://github.com/typeddjango/django-stubs#installation

**Comment by zhukovgreen** [comment]
Should be fine now

File: README.md [hunk 2]
```diff
@@ -40,6 +40,11 @@ You are also required to [configure](https://returns.readthedocs.io/en/latest/pa
 `mypy` correctly and install our plugin
 to fix [this existing issue](https://github.com/python/mypy/issues/3157):
 
+```bash
+pip install classes[compatible-mypy]
```
Comment (sobolevn) [inline README.md:44]:
```suggestion
pip install returns[compatible-mypy]
```

File: pyproject.toml [hunk 3]
```diff
@@ -48,6 +48,7 @@ _ = "returns.contrib.hypothesis._entrypoint"
 python = "^3.7"
 
 typing-extensions = ">=4.0,<5.0"
+mypy = { version = "^1.2", optional = true }
```
Comment (sobolevn) [inline pyproject.toml:51]: I think we can now remove `mypy` from `dev-dependencies`

File: README.md (hunk 2 above)
Comment (zhukovgreen) [reply README.md]: Resolved

File: pyproject.toml (hunk 3 above)
Comment (zhukovgreen) [reply pyproject.toml:51]: Why do you think so? In this case, `poetry install` won't install mypy.

Comment (zhukovgreen) [reply pyproject.toml:51]: I was checking and if we remove the mypy from the dev-dependencies, then in order to let the python installed, we need to do:
```bash
$ poetry --only main --extras compatible-mypy
```

Comment (zhukovgreen) [reply pyproject.toml]: Some small example https://asciinema.org/a/N6NpAQsKGc6T4xR6DOfDQFxDn

Comment (zhukovgreen) [reply pyproject.toml]: https://asciinema.org/a/VOKGDmAFQBA8CgaZkleKvVQs8

**Review by sobolevn — COMMENTED** [review]
Please, update mypy to `^1.4.0` and I will be happy to merge it!

**Comment by zhukovgreen** [comment]
@sobolevn seems like all done now

**Pull request merged**